### PR TITLE
Send usage stats to Grafana Loki

### PR DESCRIPTION
**Pull request by cw75**
## Describe your changes and why you are making these changes
This PR implements Go structs that can be serialized to Loki's expected log format and send it to Liki for further analysis.
The PR also adds a few new useful fields such as timestamp, environment (dev, test, or prod), status code, and log schema version number.

By default, the server runs in the "prod" mode. When we are doing dev, we should change the field via `aqueduct start --env=dev`. This will help Loki distinguish which logs are from our internal dev and which are from real users. We should modify our GitHub actions to use `aqueduct start --env=test` in the future.

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/cmd/server/server/aqueduct_server.go [hunk 1]
```diff
@@ -63,10 +63,11 @@ type AqServer struct {
 	// are no more active requests.
 	RequestMutex sync.RWMutex
 
+	Environment       string
```
Comment (kenxu95) [inline src/golang/cmd/server/server/aqueduct_server.go:66]: Can you add a short description here too?

File: src/golang/cmd/server/middleware/usage/utils.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package usage
```
Comment (kenxu95) [inline src/golang/cmd/server/middleware/usage/utils.go:1]: nit: Is there a more specific file name we can use than `utils.go`. I'm trying to get us off of dumping everything in utils.go/py if possible - have been using the term `models` for structs like these but not sure.

**Review by kenxu95 — APPROVED** [review]
Cool! Left a few non-blocking requests :)

File: src/golang/cmd/server/middleware/usage/utils.go (hunk 2 above)
Comment (cw75) [reply src/golang/cmd/server/middleware/usage/utils.go:1]: `models.go` sounds good

**Pull request merged**